### PR TITLE
(#3)Reciept generator introduced.

### LINKED ISSUE #3: Introducing the Receipt Generator.

### PR DESCRIPTION
**Pull request by vijay-singh07**
****### Resolves: #3 

### Description:-
_This PR aims to introduce an accounting module and a Reciept of invoice correponding to each transaction._

### ** HOMESCREEN**

image

### ** CREATE TRANSACTION**

image

### ** TRANSACTION LIST**

image

**EDIT TRANSACTION**

image

### **GENERATED PDF**
image

### CODE REVIEW

File: app/Http/Controllers/RegisterController.php [hunk 1]
```diff
@@ -47,8 +47,8 @@ public function loginUser(Request $request){
                 $accountant = Accountant:: where('mail','=',$request->mail)->first();
                 if($accountant){
                     if (Hash::check($request->password, $accountant->password)){
-                        $request->session()->put('loginId',$accountant->id);
-                        return redirect('dashboard');
+                        $request->session()->put('login_Id',$accountant->id);
```
Comment (Gauravbhatt19) [inline app/Http/Controllers/RegisterController.php]:
```suggestion
                        $request->session()->put('login_id',$accountant->id);
```
Use `snake_case` for session key

File: app/Http/Controllers/RegisterController.php [hunk 2]
```diff
@@ -57,7 +57,22 @@ public function loginUser(Request $request){
...
-    public function dashboard(Accountant $accountant){
-        return view('dashboard.userListing');
+    function dashboard()
+    {
+        return view('dashboard.welcome');
+    }
+
+    public function list(){
+        $accountant_data = Accountant::all();
+        return view('dashboard.userListing', ['users' => $accountant_data]);
```
Comment (Gauravbhatt19) [inline app/Http/Controllers/RegisterController.php:63]:
```suggestion
        return view('dashboard.userListing', ['users' => $accountantData]);
```
Use camelCase for laravel objects or variables or here you can simply use `users` as identifiers.

File: app/Http/Controllers/RegisterController.php (hunk 1 above)
Comment (Gauravbhatt19) [reply app/Http/Controllers/RegisterController.php]: Make similar changes throughout

File: app/Http/Controllers/TransactionController.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\transaction;
+use PDF;
+
+class TransactionController extends Controller
+{
+    public function index(){
+        return view('dashboard.createTransaction');
```
Comment (Gauravbhatt19) [inline app/Http/Controllers/TransactionController.php]:
```suggestion
        return view('dashboard.create_transaction');
```
Use snake_case to name your blade files,
Ref: https://stackoverflow.com/a/61811356

Comment (Gauravbhatt19) [reply app/Http/Controllers/TransactionController.php]: Make similar changes throughout

File: app/Http/Controllers/TransactionController.php [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            $Transaction->paid = $request->paid;
+            $Transaction->unit_amount = $request->unit_amount;
+            $Transaction->unit_quantity = $request->unit_quantity ;
+            $Transaction->unit_name = $request->unit_name;
+            $Transaction->type = $request->type;
+            $Transaction->status = $request->status;
+            $Transaction->utr = $request->utr;
+            $Transaction->comments = $request->comments;
+            $Transaction->project = $request->project;
+            $res= $Transaction-> save();
```
Comment (Gauravbhatt19) [inline app/Http/Controllers/TransactionController.php:33]:
```suggestion
            $transaction = Transaction::create($request->validated());
```
and use Request Class for validation rule.

Comment (Gauravbhatt19) [reply app/Http/Controllers/TransactionController.php:33]: similar changes throughout.

**Review by Gauravbhatt19 — COMMENTED** [review]
some minor suggestions, rest looks good to merge 👍🏻

File: app/Http/Controllers/RegisterController.php (hunk 1 above)
Comment (vijay-singh07) [reply app/Http/Controllers/RegisterController.php]: Thanks for the feedback sir, I'll update code in the latest commit.

File: app/Http/Controllers/RegisterController.php (hunk 2 above)
Comment (vijay-singh07) [reply app/Http/Controllers/RegisterController.php:63]: Thanks for the feedback sir, I'll update code in the latest commit.

File: app/Http/Controllers/TransactionController.php (hunk 3 above)
Comment (vijay-singh07) [reply app/Http/Controllers/TransactionController.php]: Thanks for the feedback sir, I'll update code in the latest commit.

File: app/Http/Controllers/TransactionController.php (hunk 4 above)
Comment (vijay-singh07) [reply app/Http/Controllers/TransactionController.php:33]: Thanks for the feedback sir, I'll update code in the latest commit.

**Review by AmanKala — APPROVED** [review]

**Pull request merged**